In [ ]:
%matplotlib inline
!pip install tensorflow tensorflow_decision_forests

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
# Load TF-DF
import tensorflow_decision_forests as tfdf
import numpy as np
import pandas as pd

import keras
from keras.datasets import mnist

In [ ]:
(x_train, y_train), (x_test, y_test) = mnist.load_data()

# Normalize Data
x_train = x_train / 255.0
x_test = x_test / 255.0

# Reshape Data
x_train = x_train.reshape(x_train.shape[0], -1)
x_test = x_test.reshape(x_test.shape[0], -1)
print(x_train.shape)

(60000, 784)


In [ ]:
model = tfdf.keras.RandomForestModel(task=tfdf.keras.Task.CLASSIFICATION, num_trees=50)
model.compile(metrics=["accuracy"])

train_data = {"feature": x_train, "label": y_train}

model.fit(x_train, y_train)

model.summary()

model.evaluate(x_test, y_test)

Use /tmp/tmpufvqgl34 as temporary training directory
Reading training dataset...
Training dataset read in 0:00:15.246110. Found 60000 examples.
Training model...
Model trained in 0:01:31.533218
Compiling model...
Model compiled.
Model: "random_forest_model_9"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
Total params: 1
Trainable params: 0
Non-trainable params: 1
_________________________________________________________________
Type: "RANDOM_FOREST"
Task: CLASSIFICATION
Label: "__LABEL"

Input Features (784):
	data:0.0
	data:0.1
	data:0.10
	data:0.100
	data:0.101
	data:0.102
	data:0.103
	data:0.104
	data:0.105
	data:0.106
	data:0.107
	data:0.108
	data:0.109
	data:0.11
	data:0.110
	data:0.111
	data:0.112
	data:0.113
	data:0.114
	data:0.115
	data:0.116
	data:0.117
	data:0.118
	data:0.119
	data:0.12
	data:0.120
	data:0.121
	data:0.122
	data:0.123
	data:0.124
	data:0.125
	data:0.126
	data:0.127
	data:0.12

[0.0, 0.9656000137329102]

Sprawozdanie:

*   Jaki współczynnik prawidłowej rozpoznawalności ma wyuczony las na zbiorze testowym?
96.56%
*   Jak wyglądają wyniki lasu decyzyjnego w stosunku do wyników sieci neuronowej
z poprzedniej listy?
96.56% w porównaniu do 97.69% z poprzedniej listy to spadek poniżej 1%, więc mozna uznać za błąd statystyczny.




